frontend/src/metabase/entities/collections.js

```diff
@@ -151,6 +151,14 @@ export const canonicalCollectionId = (
     ? null
     : parseInt(collectionId, 10);
 
+// $FlowFixMe
+export function normalizedCollection(collection) {
+  if (collection.id === null) {
+    return ROOT_COLLECTION;
+  }
+  return collection;
+}
+
```

Comment:
I believe it both accepts and returns an `ExpandedCollection`, but I couldn't get Flow to be happy and since we're planning to remove it pretty soon anyway I gave up.

Reply:
Maybe this could call `canonicalCollectionId` on the id here? I'm not sure how a collection with `id: "root"` that isn't `ROOT_COLLECTION` would end up here but it might be good to handle since I'm wary of the somewhat messy canonicalization.

Punting on trying to make Flow happy makes sense to me.

Reply:
Fair point; fixed!